### PR TITLE
feat: github actions api type with a mock implementation

### LINKED ISSUE #1520: github-actions: create a basic API that can list builds for the frontend
Make a frontend API following the current best practices.

There should be method for listing ongoing builds for a certain project for now. If possible, this can be scoped to only work for public repos in the first iteration.

### PR DESCRIPTION
**Pull request by Nek**
## Hey, I just made a Pull Request!

Added a draft Github Actions API implementation together with a API client mock. 

#### :heavy_check_mark: Checklist

- [x] All tests are passing `yarn test`
- [ ] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [x] Prettier run on changed files
- [x] Tests added for new functionality
- [ ] Regression tests added for bug fixes

### CODE REVIEW

**Review by shmidt-i — COMMENTED** [review]
Is this work towards https://github.com/spotify/backstage/issues/1520?

**Comment by Nek** [comment]
> Is this work towards #1520?

Yes

File: plugins/github-actions/package.json [hunk 1]
```diff
@@ -22,6 +22,7 @@
   },
   "dependencies": {
     "@backstage/core": "^0.1.1-alpha.13",
+    "@backstage/core-api": "^0.1.1-alpha.13",
```
Comment (Rugvip) [inline plugins/github-actions/package.json:25]: Don't use this, import from core instead. All exports from core-api are re-exported by core.

File: plugins/github-actions/src/api/GithubActionsApi.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import { createApiRef } from '@backstage/core';
+import { Build, BuildDetails } from './types';
+
+export const githubActionsApiRef = createApiRef<GithubActionsApi>({
+  id: 'plugin.githubactions.service',
+  description: 'Used by the Github Actions plugin to make requests',
+});
+
+export type GithubActionsApi = {
+  listBuilds: (_entityUri: string) => Promise<Build[]>;
```
Comment (Rugvip) [inline plugins/github-actions/src/api/GithubActionsApi.ts:26]: why the underscores here? 😅

**Review by Rugvip — COMMENTED** [review]
🎉

File: plugins/github-actions/src/api/GithubActionsApi.ts (hunk 2 above)
Comment (Nek) [reply plugins/github-actions/src/api/GithubActionsApi.ts:26]: Just an ugly leftover from the older code. Will be gone soon.

File: plugins/github-actions/package.json (hunk 1 above)
Comment (Nek) [reply plugins/github-actions/package.json:25]: Ah. I see. We still use this import on some other places.

**Pull request merged**